### DISCUSSION TITLE
I want to add/install Kibana Integrations

### DISCUSSION THREAD

**Discussion by Bajiio** [telekom-security/tpotce#1443] [Q&A]
Hello, I wanted to know if anyone already managed to add Integrations to Kibana. I currently have this error message, even if my Kibana container have access to internet: 
![image](https://github.com/telekom-security/tpotce/assets/60874928/e86ed57e-74d9-4bce-ae63-731985a49057)
I saw some of the integrations have a way to be installed by "Metricbeat" but to install this I need root rights, and of course, I don't have it. kibana user, inside kibana container, is not part of sudo group and I don't have the root password either. So if anyone has an idea, I'm open to hear it. 
Thanks in advance for your time and answer :)

**Comment by Ganesh-403** [comment]
@Bajiio 
This is most likely related to **how T-Pot packages Kibana**, rather than a permissions issue with the `kibana` user.

The **Integrations** page in Kibana is part of **Fleet**, which installs packages from the **Elastic Package Registry (EPR)** (`https://epr.elastic.co`). Even if the container has Internet access, Kibana must be able to reach the EPR and satisfy all of Fleet's prerequisites. If it cannot, you'll typically see errors when trying to browse or install integrations. :contentReference[oaicite:0]{index=0}

Regarding the lack of `sudo`:

- This is expected. T-Pot containers are intentionally hardened and generally run as non-root users.
- You are **not expected to install software (e.g. Metricbeat) from inside the Kibana container**. Container changes would also be ephemeral unless you rebuild or customize the image. The T-Pot architecture is based on Docker-managed services rather than interactive package installation. :contentReference[oaicite:1]{index=1}

A few things I'd check first:

1. **Inspect the Kibana logs** for the exact error:

   ```bash
   docker logs kibana
   ```

   Look for messages mentioning:
   - `epr.elastic.co`
   - `Fleet`
   - `Package Registry`
   - TLS/certificate or proxy errors.

2. **Verify connectivity from the container** to the Elastic Package Registry. Even if general Internet access works, a proxy, firewall, or certificate issue can prevent Kibana from reaching EPR. :contentReference[oaicite:2]{index=2}

3. **Determine whether you actually need Fleet Integrations.**

   From what I can tell, T-Pot already includes and configures the components it needs (Elasticsearch, Logstash, Kibana, Suricata, honeypots, etc.). Installing additional Fleet integrations is **not part of the documented T-Pot workflow**, so some integrations may not function as expected without additional customization. :contentReference[oaicite:3]{index=3}

If your environment is **air-gapped** or intentionally blocks access to `epr.elastic.co`, Elastic supports using a proxy or a self-hosted Elastic Package Registry for Fleet, but that requires additional configuration outside of T-Pot. :contentReference[oaicite:4]{index=4}

To better understand the issue, could you share:

- The **exact error message** shown in Kibana (a screenshot or the text).
- Your **T-Pot version**.
- Which **integration** you're trying to install (e.g. System, Windows, Suricata, etc.).

That will help determine whether this is a connectivity issue, a Fleet limitation, or simply something that isn't supported by the default T-Pot deployment.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
